test/integration/autopilot_private_firewalls/autopilot_private_firewalls_test.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package autopilot_private_firewalls
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/gcloud"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/golden"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/tft"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAutopilotPrivateFirewalls(t *testing.T) {
+	bpt := tft.NewTFBlueprintTest(t)
+	bpt.DefineVerify(func(assert *assert.Assertions) {
+		//Skipping Default Verify as the Verify Stage fails due to change in Client Cert Token
+		// bpt.DefaultVerify(assert)
+		projectId := bpt.GetStringOutput("project_id")
+		location := bpt.GetStringOutput("location")
+		clusterName := bpt.GetStringOutput("cluster_name")
+		serviceAccount := bpt.GetStringOutput("service_account")
+		clusterNetworkTag := "gke-" + clusterName
+		firewallRules := []string{"gke-%s-intra-cluster-egress", "gke-%s-webhooks", "gke-shadow-%s-all", "gke-shadow-%s-master", "gke-shadow-%s-vms", "gke-shadow-%s-inkubelet", "gke-shadow-%s-exkubelet"}
```

Comment:
Should this be?

```suggestion
		firewallRules := []string{"gke-%s-cluster-intra-cluster-egress", "gke-%s-webhooks", "gke-shadow-%s-all", "gke-shadow-%s-master", "gke-shadow-%s-vms", "gke-shadow-%s-inkubelet", "gke-shadow-%s-exkubelet"}
```

Reply:
Yes it should end up as `gke-autopilot-private-firewalls-cluster-intra-cluster-egress` but my problem is here:
I have no idea why I am trimming `clusterName` in the first place https://github.com/GorginZ/terraform-google-kubernetes-engine/blob/gke-autopilot-network-tags/test/integration/autopilot_private_firewalls/autopilot_private_firewalls_test.go#L39

Cluster name has -cluster suffix which is getting trimmed by my test for no good reason.
 https://github.com/GorginZ/terraform-google-kubernetes-engine/blob/gke-autopilot-network-tags/examples/autopilot_private_firewalls/main.tf#L38

Reply:
done https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1817/commits/8eb4e8da04c380431b05339f75ea00b15d8f9c6e